### PR TITLE
Add aliasAbstract flush test

### PR DESCRIPTION
**Pull request by TheoKouzelis**
Adds a test to prove that the aliasAbstracts property is flushed with the aliases property

### CODE REVIEW

File: src/Illuminate/Container/Container.php [hunk 1]
```diff
@@ -162,6 +162,17 @@ public function isAlias($name)
...
 
     /**
+     * Determine if a given string is an alias abstract.
+     *
+     * @param  string  $name
+     * @return bool
+     */
+    public function isAliasAbstract($name)
+    {
+        return isset($this->aliasAbstracts[$name]);
```
Comment (thecrypticace) [inline src/Illuminate/Container/Container.php]: I'm not sure this is necessary. I think that the abstract aliases should be an implementation detail not a publicly exposed thing.

You won't be able to write a test against the property but I don't think that's a bad thing. The implementation could then be changed or entirely removed if a better solution arises in the future.

**Review by thecrypticace — COMMENTED** [review]
I left some feedback. Seeing as the function seems to be the sole purpose of the PR though I'm not sure where it stands.

File: src/Illuminate/Container/Container.php (hunk 1 above)
Comment (thecrypticace) [reply src/Illuminate/Container/Container.php]: I suppose we could make the function private and reflect it during testing but that feels really, really wrong.

I'll see if I can come up with a test that fails or could in the future after a refactoring if the lists are not kept in sync.

**Comment by TheoKouzelis** [comment]
Yeah I kind of agree. 

It was the only way I could see to create a test for the one area where I thought the properties could go out of sync if you hand't added to the flush function. 

I did a search of the framework for a similar public function "isAlias". This only appears in the src/Illuminate/Container/Container.php and I would guess is only public for the purpose of testing?

**Comment by thecrypticace** [comment]
> I would guess is only public for the purpose of testing?

That's likely. It _may_ have use elsewhere but I can't think of any.

I'm trying to come up with some test case where the two out-of-sync cause an issue.

Roughly:
- There's an alias (and therefore an abstract)
- Something removing it by using `instance`
- Something else needs to be contextually built and assumes the presence of an alias when there should not be.

It _may_ be the case that it won't get tripped up but I want to verify to be sure. I've also just wrote something to ensure the two lists stay in sync. There's a slight performance penalty iff you try to remove the abstract _after_ removing the alias because I use isset on the original aliases array to be as speedy as possible. Not sure there is a way around it without yet a third array which would be overkill.

**Comment by thecrypticace** [comment]
Upon further inspection, I don't think it is possible for them being out-of-sync to be an issue (other than maybe memory — but it's such a small increase that is is essentially negligible).

For it to fail:
1. There'd have to be an alias
2. It'd have to be overwritten by an instance
3. The alias that was overwritten could _not_ be the thing being asked for. To do this one would have to nest aliases (see example below). This is a case that contextual binding still will not work on (nor do I think it is important enough to try and make work with the Container as is).

It would require walking the nested alias tree in reverse to properly support (to vend aliases — I'm not sure whether that's even a thing right now).

Otherwise when you try to vend an alias with contextual binding (`->give("alias_name_here")`) it would result in a segmentation fault due to infinite recursion.

Nested aliases:

```
        $container->instance('stub_0', new ContainerImplementationStub);
        $container->alias('stub_0', 'stub_1');
        $container->alias('stub_1', 'stub_2');
        $container->alias('stub_2', 'stub_3');
        $container->alias('stub_3', 'IContainerContractStub');
```

**Comment by thecrypticace** [comment]
If that seems like a sensible enough explanation then we can proceed with allowing this branch to be merged as-is without the need for the PR.

(thanks though for taking the time to look at this. I really appreciate it!)

**Pull request closed without merging**